### ISSUE TITLE
Provide a Flatpak distribution for Linux

### ISSUE THREAD

**Issue by gary-host-laptop** [szTheory/exifcleaner#89]
One of the biggest advantages in contrast with snaps, is that flatpaks are sandboxed, and therefore this results in a more secure and private experience since it is harder for some kind of malware to attack the system, or for said application to retrieve user information without them knowing it. Another good point is that everything that's uploaded to Flathub (Flatpak's centralized repository) is compiled by the same developers, and cannot be done by some random person on the internet which could lead to outdated packages, piracy, and whatnot. Just like snaps, Flatpaks are compatible with every distribution that supports it, so you don't have to be uploading a million different releases, and also is becoming a more standardized option for non-based and Ubuntu based distributions alike, being adopted already by twenty five flavours of Linux, and some of them come with flatpak preinstalled out of the box (CentOS, ElementaryOS, EndlessOS, Fedora, Linux Mint, PureOS and ZorinOS).

More information:

https://flathub.org/
https://flatpak.org/
https://flatpak.org/setup/
https://docs.flatpak.org/en/latest/

**Comment by szTheory** [comment]
Thanks for the suggestion! Do you know how to package Flatpak with electron-builder? I don't see it supported here https://www.electron.build/configuration/linux

**Comment by LongJohn-Silver** [comment]
> Thanks for the suggestion! Do you know how to package Flatpak with electron-builder? I don't see it supported here https://www.electron.build/configuration/linux

No, sorry, I am not a developer and I lack the technical knowledge, but I know that Element is uses an Electron wrap up of their web service and provide a flatpak, Signal and Jitsi's desktop client is also built with Electron and they, too, provide flatpaks s I guess there must be a work around.

I found these links which may be of help, but I think the best solution would be to use the flatpak builder, although I don't know exactly.

https://github.com/electron-userland/electron-builder/issues/512
https://github.com/endlessm/electron-flatpak-base-app

**Comment by szTheory** [comment]
Thanks, it sounds like electron-builder might add support for FlatPak eventually. Users are contributing to a bounty for it in the issue you linked to electron-userland/electron-builder#512. Once electron-build adds support I'll probably add it to ExifCleaner too. I'm moving the issue to "blocked" until then to revisit it at a later date.

**Comment by LongJohn-Silver** [comment]
There has been some updates, if not with regards to the issue being solved, but rather with some documentation that could help as a work around. Maybe you could check it out and see if it is doable, since I lack the technical skills.

**Comment by LongJohn-Silver** [comment]
@szTheory It has been fixed. : )

**Comment by szTheory** [comment]
Thanks! I moved this back into the backlog of things to do. I've been having trouble keeping up with the manual release process so I'm working on putting together a CI/CD workflow to help with automated packaging and testing of installs across every permutation of OS and package type. That will make it easier to slot changes like this in. It will probably be a while still though.

**Comment by szTheory** [comment]
I just checked out https://github.com/electron-userland/electron-builder/issues/512 and it looks like Flatpak support in electron-builder is still experimental, so I'm going to hold off for now since I wouldn't be able to support it until it's stable. I'll keep this open to follow up on it periodically, though. Thanks again.

**Comment by szTheory** [comment]
Triage note, so this thread isn't silently sitting open: labeled `maybe-someday` and `help wanted`. **Not scheduled**, and not closed either.

The release pipeline currently builds an exact, gated inventory of seven artifacts (`scripts/release_asset_gate.mjs:12-24`): two DMGs, the NSIS installer, the Windows portable exe, an AppImage, a deb and an rpm. Adding a Flatpak isn't a one-time change — it's a channel that has to keep working on every future release, and I'm not in a position to promise that upkeep.

What would move it: a PR from someone who already packages for Flathub, or an independently maintained manifest I can link from the README. Either is welcome and I'll review it.

**Comment by szTheory** [comment]
Closing as not planned — which reverses the note I left on this thread earlier today saying I'd leave it open. On reflection, keeping it open for another four years isn't more honest than saying plainly where it stands.

Where it stands: I don't have the time to take on a distribution channel that then has to keep working on every future release, without help. That isn't a judgment about Flathub — it's that the upkeep is real and ongoing, while demand for this particular channel has stayed quiet.

Both halves of that can change, and either one flips it:

- **Demand.** If people actually want ExifCleaner on Flathub, say so here. Comments and 👍 on this issue are exactly the signal I'd act on.
- **Help.** If you already package for Flatpak, I'll give guidance, review a PR, and sort out the release-pipeline side with you. I just can't start it and carry it alone.

Reopening costs nothing — a comment here does it.
